### DISCUSSION TITLE
how to implement fix/disabled item with Sortable

### DISCUSSION THREAD

**Discussion by TianboZ** [clauderic/dnd-kit#855] [Q&A]
https://user-images.githubusercontent.com/28853847/182753436-94f9f319-d93d-44a3-94d2-8e63292f4803.mp4

when i drag over item 2 and move it up, it squeezes item 1 to the position of item 2. How can I get a fixed item that its position can't  be squeezed away when dragging the item over it? i tried to set `disabled` with `useSortable`, but it does not help. I found a trick to do this, see below image, but not sure if there is better way.

![image](https://user-images.githubusercontent.com/28853847/182754014-2131a8fc-12d5-4784-bef6-0f80a753a4dc.png)

**Comment by aryan-mehrabi** [comment]
@TianboZ Did you ever manage to achieve this?

**Reply by robahtou** [reply]
ran into the same issue. `disabled: true` is not working for a particular `id`

**Reply by aryan-mehrabi** [reply]
@robahtou I don't know if this suits your case but you can pass `setNodeRef` conditionally like this:

```js
<div
  {...attributes}
  ref={isDisabled ? undefined : setNodeRef}
>
...
</div>
```

or if you're using `setDroppableNodeRef`:

```js
<div
  {...attributes}
  ref={isDisabled ? undefined : setDroppableNodeRef}
>
...
</div>
```

**Reply by hsavit1** [reply]
@aryan-mehrabi this is a great hack, thanks for sharing!

**Reply by btrianurdin** [reply]
thank you man! it's work...

**Reply by pranavdhamanage-work** [reply]
Did anyone find a proper way to fix it? From what I have understood this needs to accomplished by updating vertical list sorting strategy, handling reordering of items in the onDragEnd handler and maybe the updation of collision strategy as well

**Reply by pranavdhamanage-work** [reply]
@clauderic do you have insights on this? How can this be achieved properly because this seems to be a common use case.

**Reply by MrDoomBringer** [reply]
I saw this solved with a custom sorting strategy, which worked for me - https://gist.github.com/MrDoomBringer/e49dee9a0743f42d19879e05b9b628ee
